testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/test_sgreplicate_cbl.py

```diff
@@ -0,0 +1,2139 @@
+        replication_type="push_pull", continuous=True)
+
+    # Create docs with attachments
+    db.create_bulk_docs(channel1_docs, Replication1_channel1, db=cbl_db1, channels=channels1)
+    replicator.wait_until_replicator_idle(repl3)
+    sg_docs = sg_client.get_all_docs(url=sg1.url, db=sg_db1, auth=session)["rows"]
+    db.create_bulk_docs(channel2_docs, Replication1_channel2, db=cbl_db1, channels=channels2, attachments_generator=attachment.generate_png_100_100)
+    channel3_doc_ids = db.create_bulk_docs(channel3_docs, Replication1_channel3, db=cbl_db1, channels=channels3, attachments_generator=attachment.generate_png_100_100)
+
+    # 4. pushpull replication from sg1 -> sg2
+    repl_id_1 = sg1.start_replication2(
+        local_db=sg_db1,
+        remote_url=sg2.url,
+        remote_db=sg_db2,
+        remote_user=name4,
+        remote_password=password,
+        direction="pushAndPull",
+        channels=channels1,
+        continuous=continuous
+    )
+    # 4. verify docs with channel1 which is filtered in replication should get replicated to cbl_db2
+    sg1.admin.wait_until_sgw_replication_done(sg_db1, repl_id_1, write_flag=True)
+    # Do pull replication from sg2 -> cbl2
+    repl4 = replicator.configure_and_replicate(
+        source_db=cbl_db2, replicator_authenticator=replicator_authenticator4, target_url=sg2_blip_url,
+        replication_type="push_pull", continuous=True)
+    # Verify docs with attachments are replicated to sgw cluster2
+    sg_docs_attachments = sg_client.get_all_docs(url=sg1.url, db=sg_db1, auth=session, include_docs=True)["rows"]
+    for doc in sg_docs_attachments:
+        if doc["id"] in channel3_doc_ids:
+            assert "_attachments" in doc["doc"], "attachment did not replicated on sgw cluster 2"
+
+    # update docs by adding attachments
+    for doc in sg_docs:
+        sg_client.update_doc(url=sg1.url, db=sg_db1, doc_id=doc["id"], number_updates=1, auth=session, attachment_name=file_attachment)
+    # update docs by deleting attachments
+    db.update_bulk_docs_by_deleting_blobs(cbl_db1, doc_ids=channel3_doc_ids)
+
+    # wait until replication completed
+    replicator.wait_until_replicator_idle(repl3)
+    replicator.wait_until_replicator_idle(repl4)
+    # Verify docs created in cbl2
+    cbl_doc_ids2 = db.getDocIds(cbl_db2)
+    count = sum(Replication1_channel1 in s for s in cbl_doc_ids2)
+    assert count == channel1_docs, "all docs with channel1 did not replicate to cbl db2"
+    count = sum(Replication1_channel3 in s for s in cbl_doc_ids2)
+    assert count == channel3_docs, "all docs with channel3 did not replicate to cbl db2"
+
+    # 5. Verify docs with channel2 is not accessed by user 2 i.e cbl db2
+    count = sum(Replication1_channel2 in s for s in cbl_doc_ids2)
+    assert count == 0, "all docs with channel2 replicated to cbl db2"
+
+    # 6. Verify all docs updated with attachments
```

Comment:
summary can not match the steps in this test case, could you please refine them?